ToolBox/TickAggregatingConsolidator.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.ToolBox
+{
+    /// <summary>
+    /// Class that uses consolidators to aggregate tick data data
+    /// </summary>
+    public abstract class TickAggregatingConsolidator
+    {
+        /// <summary>
+        /// The consolidator used to aggregate data from
+        /// higher resolutions to data in lower resolutions
+        /// </summary>
+        public abstract IDataConsolidator Consolidator { get; }
+
+        /// <summary>
+        /// The consolidated data
+        /// </summary>
+        public List<BaseData> Consolidated { get; protected set; }
+
+        /// <summary>
+        /// The resolution that the data is being aggregated into
+        /// </summary>
+        public Resolution Resolution { get; protected set; }
+
+        /// <summary>
+        /// This method is intended to return all of the
+        /// aggregated data and the bar that the consolidator is working on
+        /// </summary>
+        public abstract List<BaseData> Flush();
```

Comment:
Why does this method need to be abstract? It appears all implementations do the same thing save the unnecessary cast to a specific type (`TradeBar` vs `QuoteBar`)